projects/native-lend/index.js

```diff
@@ -1,26 +1,53 @@
-
-const { compoundExports2 } = require("../helper/compound");
-const { lendingMarket } = require("../helper/methodologies");
+const { getLogs } = require("../helper/cache/getLogs");
+const { sumTokens2 } = require("../helper/unwrapLPs");
 
 module.exports = {
-  methodology: lendingMarket
+  methodology: "Gets all the assets deposited by LPs in Native Credit Pool for PMMs to facilitate trades for Native Swap.",
 };
 
 const config = {
-  arbitrum: "0x795E5cCdd008637EB98c64958d3b8Ba3e6DE2d2B",
-  zklink: "0x4AC97E2727B0e92AE32F5796b97b7f98dc47F059",
-  bsc: "0x3ba16AC2A67D126BF1DBa0a81E6C75073EFd95d9",
-  mantle:"0x48A6FE0Fa8DfF6D290Bd21aB6BCC1DDAeb9f2D0e",
-  ethereum:"0x3ba16AC2A67D126BF1DBa0a81E6C75073EFd95d9",
-  zeta:"0x3ba16AC2A67D126BF1DBa0a81E6C75073EFd95d9"
+  ethereum: {
+    vault: "0xe3D41d19564922C9952f692C5Dd0563030f5f2EF",
+    vaultFromBlock: 22173196,
+  },
+  bsc: {
+    vault: "0xBA8dB0CAf781cAc69b6acf6C848aC148264Cc05d",
+    vaultFromBlock: 47980948,
+  },
+  base: {
+    vault: "0x097534f09EB81Cc5B69aDF36df3d3fA11b7FE1B1",
+    vaultFromBlock: 28407972,
+  },
+  arbitrum: {
+    vault: "0x273EcF5BDac92bae19ea2B525EA99F54108F03bd",
+    vaultFromBlock: 355259594,
+  },
 };
 
 Object.keys(config).forEach((chain) => {
-  const comptroller = config[chain];
-  module.exports[chain] = compoundExports2({ comptroller })
-  module.exports[chain].tvl = async (api) => {
-    const markets = await api.call({ abi: 'address[]:getAllMarkets', target: comptroller })
-    const tokens = await api.multiCall({  abi: 'address:underlying', calls: markets})
-    return api.sumTokens({ owner: comptroller, tokens })
-  }
-})
\ No newline at end of file
+  // get vault and start block
+  const { vault, vaultFromBlock } = config[chain];
+  module.exports[chain] = {
+    tvl: async (api) => {
+      // get all lps
+      const lpListedLogs = await getLogs({
+        api,
+        target: vault,
+        topic: "MarketListed(address)",
+        eventAbi:
+          "event MarketListed(address lpToken)",
+        onlyArgs: true,
+        fromBlock: vaultFromBlock,
+      });
+      const lps = lpListedLogs.map((i) => i.lpToken);
+
+      // get all underlying tokens
+      const tokens = await api.multiCall({ abi: 'address:underlying', calls: lps });
+
+      // get total supplied, not cash
```

Comment:
`tvl` of lending protocols should be available liquidity. Please export any borrows as `borrows` so that users can choose to include them in TVL stats